### PR TITLE
[enhancement] Diagnostic script to monitor system level data during a restore point operation

### PR DESCRIPTION
**Pull request by rakrashba**
During snapshot operations on Linux we often come across bugs that are pretty tricky to debug without any knowledge about the system state at that point in time. This PR will introduce a small app written in go that will do exactly that.

The data this app will capture can be considered customer data - especially when strace is on - hence this app will not be shipped with normal extensions code.

The README has extensive directions on how to build and run this both manually and as part of a Restore Point operation.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -97,6 +97,8 @@ VMBackup/MANIFEST
 VMBackup/*.pyproj.user
 VMBackup/*.suo
 VMBackup/main/safefreeze/bin/*
+# VMBackup debughelper builds
+VMBackup/debughelper/msft_snap_monit
```
Comment (iamashwani-who) [inline .gitignore:101]: is this the output path for binary compilation?

File: VMBackup/main/handle.py [hunk 2]
```diff
@@ -582,7 +621,8 @@ def daemon():
         backup_logger.commit_to_local()
         if(eventlogger is not None): 
             eventlogger.dispose()
-
+    if monitor_process is not None:
+        monitor_process.terminate()
```
Comment (iamashwani-who) [inline VMBackup/main/handle.py:625]: line 588 uses os.kill while we use process.terminate here. Why is the difference?

File: .gitignore (hunk 1 above)
Comment (rakrashba) [reply .gitignore:101]: Yes. The binary file is by default created in the working directory itself. The name of the binary is defined by the module name which can be found in the first line of `VMBackup/debughelper/go.mod`

**Review by iamashwani-who — DISMISSED** [review]

**Review by vityagi — APPROVED** [review]
Approved

**Review by iamashwani-who — APPROVED** [review]

**Review by ankitha-p — APPROVED** [review]

**Comment by rakrashba** [comment]
Merging.

**Pull request merged**